client/file-transfer-service/src/handler.rs

```diff
@@ -151,6 +152,73 @@ impl Actor for FileTransferService {
+
+                FileTransferServiceCommand::DownloadResponse {
+                    request_id,
+                    file_key_proof,
+                    callback,
+                } => {
+                    let response =
+                        schema::v1::provider::response::Response::RemoteDownloadDataResponse(
+                            schema::v1::provider::RemoteDownloadDataResponse {
+                                file_key_proof: file_key_proof.encode(),
+                            },
+                        );
+
+                    let mut response_data = Vec::new();
+                    response.encode(&mut response_data);
+
+                    let outgoing_response = OutgoingResponse {
+                        sent_feedback: None,
+                        result: Ok(response_data),
+                        reputation_changes: Vec::new(),
+                    };
+
+                    let maybe_pending_response =
+                        self.download_pending_responses.remove(&request_id).take();
+
+                    match maybe_pending_response {
+                        Some(pending_response_sender) => {
+                            let pending_response_result =
+                                pending_response_sender.send(outgoing_response);
+                            match pending_response_result {
+                                Ok(()) => {
+                                    match callback.send(Ok(())) {
+                                        Ok(()) => {}
+                                        Err(_) => error!(
+                                            target: LOG_TARGET,
+                                            "Failed to send the response back. Looks like the requester task is gone."
+                                        ),
+                                    };
+                                }
+                                Err(e) => {
+                                    error!(
+                                        target: LOG_TARGET,
+                                        "Failed to return Download Response {:?}", e
+                                    );
+                                    match callback.send(Err(RequestError::ResponseFailure(e))) {
+                                        Ok(()) => {}
+                                        Err(_) => error!(
+                                            target: LOG_TARGET,
+                                            "Failed to send the response back. Looks like the requester task is gone."
+                                        ),
+                                    };
+                                }
+                            }
+                        }
+                        None => {
+                            error!(target: LOG_TARGET, "No pending response channel found for request id {:?}", request_id);
+                            match callback.send(Err(RequestError::DownloadRequestIdNotFound)) {
```

Comment:
This `match callback.send` is replicated 3 times. I think it would be more readable if you do it only once at the end.

Reply:
Thanks, I wasn't careful enough here. I improved it!